.github/workflows/release.yml

```diff
@@ -72,12 +72,32 @@ jobs:
       - name: Build dependencies
         run: pnpm run build
 
+      - name: Checkout main repository
+        if: ${{ inputs.ref && inputs.ref != 'main' }}
+        uses: actions/checkout@v5
+        with:
+          path: 'main'
+
+      - name: Release packages
```

Comment:
Skipped steps are still shown on the UI on action runs, so this and the following steps should probably have different names (maybe a " (main)" suffix)